### PR TITLE
Add how-to for replacing a storage node on Exoscale

### PR DESCRIPTION
**Pull request by simu**
Tested 5 times so far.

Overall, the instructions work reliably now. MON replacement is a bit ugly at the moment, due to the operator not picking up health check config changes unless it's restarted.

With the current instructions and `removeOSDsIfOutAndSafeToRemove: true` in the `CephCluster` CR, the operator will sometimes clean up the OSD deployment quickly, but not always. Therefore (and to cover clusters with `removeOSDsIfOutAndSafeToRemove: false`), the how-to contains steps to delete the OSD deployment and PVC.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+* You want to replace an existing storage node in the cluster with a new storage node
+
+== Prerequisites
+
+The following CLI utilities need to be available locally:
+
+* `exo` >= v1.28.0 https://community.exoscale.com/documentation/tools/exoscale-command-line-interface[Exoscale CLI]
+* `vault` https://www.vaultproject.io/docs/commands[Vault CLI]
+* `commodore`, see https://syn.tools/commodore/running-commodore.html[Running Commodore]
+
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc:21]:
```suggestion
* `commodore`, see https://syn.tools/commodore/running-commodore.html[Running Commodore]
* `jq`
```

File: docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
++
+[source,bash]
+----
+kubectl --as=cluster-admin delete pv "${pv_name}"
+----
+
+=== Finish up
+
+. Re-enable ArgoCD auto sync
++
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc:532]: There doesn't seem to be a step that says to disable ArgoCD sync?

Comment (simu) [reply docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc:532]: It doesn't say "Disable ArgoCD sync", but "Disable auto sync for component `rook-ceph`": https://github.com/appuio/openshift4-docs/pull/98/files#diff-20fce70a1a58209ea5e6297cd08cbbcf33acac367617ce7cb9670539bba0ab1aR203

Comment (ccremer) [reply docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc:536]: ah, missed that one and I was doing a browser search for "ArgoCD" :)

**Review by corvus-ch — CHANGES_REQUESTED** [review]
Please look into ways to not break somebodies exoscale configuration.

File: docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+  xargs oc --as=cluster-admin adm certificate approve
+
+kubectl --as=cluster-admin get nodes
+----
+
+. Label and taint the new storage node
++
+[source,bash]
+----
+kubectl --as=cluster-admin label --overwrite node -lnode-role.kubernetes.io/worker \
```
[the remark this answers is not in the document]
Comment (simu) [reply docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc]: done for a few bits that I was aware are duplicated.

File: docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+----
++
+include::partial$vshn-input.adoc[]
+
+. Configure Exoscale CLI
++
+[source,bash]
+----
+mkdir -p ~/.config/exoscale
+cat <<EOF >> ~/.config/exoscale/exoscale.toml
```
[the remark this answers is not in the document]
Comment (simu) [reply docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc]: Turns out this is not required at all when configuring a recent `exo` cli with environment variables. I've removed this here and in the install docs.

File: docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+
+* You already have a OpenShift 4 cluster on Exoscale
+* You have admin-level access to the cluster
+* You want to replace an existing storage node in the cluster with a new storage node
+
+== Prerequisites
+
+The following CLI utilities need to be available locally:
+
+* `docker`
```
[the remark this answers is not in the document]
Comment (simu) [reply docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc:21]: Required to run Terraform in Docker.

File: docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc [hunk 6]
```diff
@@ -0,0 +1,556 @@
...
+* `curl`
+* `kubectl`
+* `oc`
+* `exo` >= v1.28.0 https://community.exoscale.com/documentation/tools/exoscale-command-line-interface[Exoscale CLI]
+* `vault` https://www.vaultproject.io/docs/commands[Vault CLI]
+* `commodore`, see https://syn.tools/commodore/running-commodore.html[Running Commodore]
+* `jq`
+* `yq` https://mikefarah.gitbook.io/yq[yq YAML processor] (version 4 or higher)
```
[the remark this answers is not in the document]
Comment (simu) [reply docs/modules/ROOT/pages/how-tos/exoscale/replace_storage_node.adoc:29]: Required to extract Terraform Docker image version (cf. `docs/modules/ROOT/partials/setup_terraform.adoc`)

**Review by corvus-ch — APPROVED** [review]

**Review by ccremer — APPROVED** [review]

**Pull request merged**